jetty-http/src/main/java/org/eclipse/jetty/http/HttpCompliance.java

```diff
@@ -18,179 +18,394 @@
-    RFC2616_LEGACY(sectionsBySpec("RFC2616,-FIELD_COLON,-METHOD_CASE_SENSITIVE,-TRANSFER_ENCODING_WITH_CONTENT_LENGTH,-MULTIPLE_CONTENT_LENGTHS")), 
-    
-    /** The strict RFC2616 support mode */
-    RFC2616(sectionsBySpec("RFC2616")), 
-    
-    /** Jetty's current RFC7230 support, which incorrectly excludes  {@link HttpComplianceSection#METHOD_CASE_SENSITIVE} */
-    RFC7230_LEGACY(sectionsBySpec("RFC7230,-METHOD_CASE_SENSITIVE")),
-
-    /** The RFC7230 support mode */
-    RFC7230(sectionsBySpec("RFC7230")),
-    
-    /** Custom compliance mode that can be defined with System property <code>org.eclipse.jetty.http.HttpCompliance.CUSTOM0</code> */
-    @Deprecated
-    CUSTOM0(sectionsByProperty("CUSTOM0")),
-    /** Custom compliance mode that can be defined with System property <code>org.eclipse.jetty.http.HttpCompliance.CUSTOM1</code> */
-    @Deprecated
-    CUSTOM1(sectionsByProperty("CUSTOM1")),
-    /** Custom compliance mode that can be defined with System property <code>org.eclipse.jetty.http.HttpCompliance.CUSTOM2</code> */
-    @Deprecated
-    CUSTOM2(sectionsByProperty("CUSTOM2")),
-    /** Custom compliance mode that can be defined with System property <code>org.eclipse.jetty.http.HttpCompliance.CUSTOM3</code> */
-    @Deprecated
-    CUSTOM3(sectionsByProperty("CUSTOM3"));
-  
     public static final String VIOLATIONS_ATTR = "org.eclipse.jetty.http.compliance.violations";
 
-    private static final Logger LOG = Log.getLogger(HttpParser.class);
-    private static EnumSet<HttpComplianceSection> sectionsByProperty(String property)
+    // Identification of the mode
+    private String mode;
+    // The default values here are just Jetty defaults and not associated with any specific RFC/Spec
+    // Use the appropriate builder to start with a specific RFC/Spec behavior
+    private boolean allowCaseInsensitiveFieldNames = false;
+    private boolean allowCaseInsensitiveRequestMethod = true;
+    private boolean allowHttp09 = false;
+    private boolean allowMultiLineFieldValue = false;
+    private boolean allowMultipleContentLengths = false;
+    private boolean allowTransferEncodingWithContentLength = false;
+    private boolean allowWhitespaceAfterFieldName = false;
+    private boolean requireColonAfterFieldName = true;
+    private boolean useCaseInsensitiveFieldValueCache = true;
+
+    // Prevent direct instantiation
+    private HttpCompliance()
     {
-        String s = System.getProperty(HttpCompliance.class.getName()+property);
-        return sectionsBySpec(s==null?"*":s);
     }
 
-    static EnumSet<HttpComplianceSection> sectionsBySpec(String spec)
+    /**
+     * Get the Default HttpCompliance of <a href="https://tools.ietf.org/html/rfc7230">RFC 7230: HTTP/1.1</a>
+     *
+     * @return default HttpCompliance configuration of RFC7230 Strict
+     */
+    public static HttpCompliance getDefault()
     {
-        EnumSet<HttpComplianceSection> sections;
-        String[] elements = spec.split("\\s*,\\s*");
```

Comment:
For backwards compatibility and easy creation of configuration modes from XML, we should still support building a HttpCompliance from a spec String